### DISCUSSION TITLE
TouchedFields not updating as expected

### DISCUSSION THREAD

**Discussion by keisto** [react-hook-form/react-hook-form#9586] [Q&A] [answered]
### Version Number

7.40.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-controller-forked-2rd6c8?file=/src/index.js:0-2005

### Steps to reproduce

1. Go to 'https://codesandbox.io/s/react-hook-form-v7-controller-forked-2rd6c8?file=/src/index.js:0-2005'

// Nested checkbox
2. Click on 'checkbox'
3. Click 'submit'
4. Check console for "touchedFields"

// FieldArray move
5. Select left|right buttons to move around A,B,C
6. Click 'submit'
7. Check console for "touchedFields"

### Expected behaviour

// Nested checkbox
I would expect touchedFields to include `testNestedBool.isChecked`

// FieldArray move
I would expect touchedFields to include `testFieldArray`

### What browsers are you seeing the problem on?

Chrome, Safari

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
simple one: missing `onBlur` event handler

**Reply by bluebill1049** [reply]
![Screenshot 2022-12-14 at 9 03 59 am](https://user-images.githubusercontent.com/10513364/207453854-412e039f-a66a-45ab-bf74-eeaac39b3c77.png)

https://react-hook-form.com/api/usecontroller

**Reply by keisto** [reply]
Updated. Okay this handles the checkbox but only for Chrome and Firefox. This doesn't work in Safari. (only tested these 3 browsers)

**Reply by bluebill1049** [reply]
https://bugs.webkit.org/show_bug.cgi?id=13724

**Reply by keisto** [reply]
Perfect, that makes sense then.

I also put in an additional issue with this, the useFieldArray - `move()`. I expected that field to be added to the touchedFields after moving

**Reply by keisto** [reply]
Marking as answered, will create new discussion for useFieldArray `move()`.
